tests/kernel/mem_heap/mheap_api_error_case/src/test_mheap_error.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+/* action for self-defined assert handler. */
+static ZTEST_DMEM volatile bool valid_assert;
+
+static inline void set_assert_valid(bool valid)
+{
+	valid_assert = valid;
+}
+
+static void action_after_assert_fail(void)
+{
+	ztest_test_pass();
+}
+
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
+#endif
+{
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+	ARG_UNUSED(file);
+	ARG_UNUSED(line);
+#endif
+
+	printk("Caught an assert.\n");
+
+	if (valid_assert) {
+		valid_assert = false; /* reset back to normal */
+		printk("Assert error expected as part of test case.\n");
+
+		/* do some action after fatal error happened */
+		action_after_assert_fail();
+	} else {
+		printk("Assert failed was unexpected, aborting...\n");
+#ifdef CONFIG_USERSPACE
+	/* User threads aren't allowed to induce kernel panics; generate
+	 * an oops instead.
+	 */
+		if (_is_user_context()) {
+			k_oops();
+		}
+#endif
+		k_panic();
+	}
+}
+
+/**
+ * @brief Test Test input aligen value zero
```

Comment:
Double "Test". Please remove one.

Reply:
I've changed it. Thank you for your careful review.